DSharpPlus.UnifiedCommands/Message/Internals/MessageFactory.cs

```diff
@@ -0,0 +1,314 @@
+using System.Diagnostics;
+using System.Text;
+using DSharpPlus.UnifiedCommands.Message.Conditions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace DSharpPlus.UnifiedCommands.Message.Internals;
+
+internal class MessageFactory
+{
+    private readonly MessageTree _commands = new();
+    private readonly List<Func<IServiceProvider, IMessageCondition>> _messageConditionBuilders = new();
+    private readonly IServiceProvider _services;
+
+    internal void AddCommand(string name, MessageMethodData data) => _commands.Branches!.Add(name, new(data));
+    internal void AddBranch(string name, MessageTree branch) => _commands.Branches!.Add(name, branch);
+
+    internal MessageTree? GetBranch(string name) =>
+        _commands.Branches!.TryGetValue(name, out MessageTree? result) ? result : null;
+
+    public MessageFactory(IServiceProvider services)
+        => _services = services;
+
+    internal void AddMessageConditionBuilder(Func<IServiceProvider, IMessageCondition> func)
+        => _messageConditionBuilders.Add(func);
+
+    internal void ConstructAndExecuteCommand(Entities.DiscordMessage message,
+        DiscordClient client, ref ReadOnlySpan<char> args, List<Range> argsRange)
+    {
+        long startTime = Stopwatch.GetTimestamp();
+
+        Index end = 0;
+        int it = 0;
+
+        MessageTree? tree = null;
+
+        foreach (Range argRange in argsRange)
```

Comment:
this entire parsing logic is very much not how we should parse things. set a todo to change it, i can deal with it later